File: src/main/java/com/automation/pages/HomePage.java
package com.automation.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

public class HomePage extends BasePage {

    // "Log In" link in the top navigation of try.frndlytv.com
    @FindBy(xpath = "//a[normalize-space(text())='Log In' or normalize-space(text())='Sign In']"
            + " | //button[normalize-space(text())='Log In' or normalize-space(text())='Sign In']")
    private WebElement loginButton;

    public HomePage(WebDriver driver) {
        super(driver);
    }

    public FrndlyLoginPage clickLogin() {
        click(loginButton);
        return new FrndlyLoginPage(driver);
    }
}

File: playwright/pages/HomePage.ts
import { Page } from '@playwright/test';
import { BasePage } from './BasePage';
import { FrndlyLoginPage } from './FrndlyLoginPage';

export class HomePage extends BasePage {
  constructor(page: Page) {
    super(page);
  }

  async clickLogin(): Promise<FrndlyLoginPage> {
    const loginBtn = this.page.locator(
      "//a[contains(@href, '/authenticator')] | //button[normalize-space(text())='Log In'] | //button[normalize-space(text())='Sign In']"
    ).first();
    await loginBtn.waitFor({ state: 'visible' });
    await loginBtn.click();
    return new FrndlyLoginPage(this.page);
  }
}
